### ISSUE TITLE
20 bugs

### ISSUE THREAD

**Issue by rickparrish** [rickparrish/Usurper#10]
Might make more sense to split these up into individual issues, but I'm being lazy and pasting under one issue for now.

This is from a post on Jakob's Usurper forum by Haakon, and other follow-up posts.

1. The bank can be robbed over and over, especially when macros are set, and a player be able to raise tens of levels in short order.

2. Going home has no balance; the go to sleep option should just be temporarily disabled until you can break into someone's home somehow, in which case
the inn doorguards should be available at one's home, and one shouldn't be able to just go home, without having first purchased one; especially not be
able to marry before buying a home. the homes should be editable and their prices and possible bonus attributes (extra hit points, armor points, etc. when
defending better homes).

3. The monster editor is almost useless. None of the monster settings you change are actually recognized by Usurper except for strength. Strength isn't
even strength, it's strength and hitpoints. Higher strength settings results in harder hits and more monster hitpoints. Having all the monsters accurately
use all the fields would be awesome. 

Dan: hitpoints and defense are honoured now. The other fields (intelligence, etc.) are not. I think they should be removed.

4. Many of the spells don't work properly, such as a sage's escape spell, and many others. The ones that do are often very weak. When can a spell that
hits for 50 hitpoints really ever be all that useful. They should be able to be configured to hit for a certain average, and perhaps be able to specify
the standard deviation of the spread...that would be nice...

Dan: the existing spells are fixed, but the comment on them being weak is still applicable.

5. Having the Supreme Being's stats based on some sort of percentage of whatever, and being hard coded, tends to make him weak, especially in customized
games. Being able to atleast toggle between that or to the game recognizing his stats like all the other monsters. 

6. Teammates you bring to the dungeon all die incredibly fast after increasing monsters' strength. By resurrecting his teammates and fighting in the dungeon,
the person can kill many more monsters than ever intended, since his teammates instantly die, leaving all the monsters that were meant for them there for
him to kill alone.

7. The experience lost when losing in a player fight should not be tied to monster percentages, and should be independently configurable. You may have
a tough dungeon and want no experience lost, but want players to lose experience in a player fight. Lastly, you should be able to specify fractions of
percents...to me, 1% is even often too much to use, so being able to say .25 would be great. 

Dan: fractions of percents are now supported.

8. Fighting doorguards can sometimes result in it missing completely when it tries to hit you. Why does this only occur here? Shouldn't that happen everywhere
based on your agility? 

9. The \"range\" when selecting a moat monster goes from \"0 .. 8, please\" even when you have more than 8 monsters, making all after 8 inaccessible. 

Dan: I think Jake fixed this in his last version.

10. Challenging a team at times when they are all busy takes a fight from you even though you didn't actually get to fight. 

Dan: fixed.

11. You can be dead and not allowed to fight when your team gets attacked, but be allowed to defend the gym. Seems like you shouldn't be able to defend
the gym. 

Dan: agreed - fixed.

12. It would be nice to be able to set a max number of times you can play hi-low. 

13. There is a bug with the inventory/equip/unequip code. You can't properly specify which ring (and maybe neck too) finger you're wanting to add to, resulting
in you having to remove both items, and then add the two you want to use. 

Dan: partially fixed; still working on this.

14. Children should give you more daily experience. Children should also be able to do other neat things, for example, the square root of the number of
children you have is that many bonus dungeon fights you get, etc. Children would be very useful, and then should start costing a lot to keep. That should
be able to be configured too -- cost per child per day perhaps.

15. Cost of being married should be configurable. Being married should also do more than just let you force an intimate session. The intimate sessions themselves
perhaps could maybe be configurable, allowing you to make having sex give you much more experience than it does now, and reduce the number of times you
can do it. Who can really have sex 20 times in a day...like when two people use their 10 turns on each other? 

16. Many of the delay prompts take too long and could really be pause prompts (having sex takes forever). 

Dan: I think Jake fixed this in his last version.

17. There is no balance to being evil. You must be evil to really get ahead. Being evil earns you up to 6 times stealing beer. Those who play a good character
have no way of having their own kind of edge, since being good gives you only 3 times stealing beer, although I have somewhat countered this by offering
certain items cheaper and only available to the good characters. 

Dan: stealing gives half as much exp now, but the imbalance is still present.

18. It would be nice if when you wanted to rape the girl in the dungeon, you didn't end up \"saving\" her and I think earning chivalry when you killed
the hostile monsters. 

Dan: fixed.

19. Beer drinking can be done over and over.

Dan: did Jake fix this??

20. There is a major flaw I won't even actually reveal for I don't want rampant cheating to occur, because the bug allows you to duplicate items and gain
lots of money.

Dan: I really want to know how this works! Someone else also said it:

2) Also a major bug still exists..the money/healing potion duplication trick that has been around for years..I have a character on a board with 2,000,000,000
healings do to this bug...email me for further explanation of where the bug is located..

Dan: would really like to fix this.

**Comment by HaakonOfTheVortex** [comment]
Wow, I forgot I had ever written this!  We just ported the Vortex over to Linux and one of our last things to do is go back and recreate our Ultimate Usurper, where we had added a bunch of items and changed levels and monsters and all sorts of fun stuff.  Through all the hours of testing making sure everything was in sync, and years of playing, we identified lots of things like these issues.  I'm so glad it's helping drive Usurper forward.  Arelas and I would be more than happy to help you test bug fixes or even be an experimental Usurper site.  Since putting the board back up a couple of years ago we've been slow to advertise it and start driving up traffic, mainly because we had some things we wanted to finish before really pushing it.  However we are pretty much to that point now and need to start letting people know we're back up for good.  You can check out our website at http://vortexbbs.com.  Anyway, I'll see if I can figure out how to send you a private message about but #20 but damn if I can remember how I did #2 with the healing potions!  Anyway, so glad to see you are still working on this.  Anything we can do to help we'll gladly do!  Thanks!

**Comment by dan1982code** [comment]
Thank you! Any further information about these bugs is valuable. I knew the codebase quite well 4 years ago when I was really pushing on this, but do remember that the bugs were often not all that hard to fix. The biggest problem with the code was the extensive code duplication all over the place. Huge chunks of code were duplicated multiple times -- and when a bug was fixed in one of the copies, it often wasn't fixed in others. I and Rick did a lot of code cleanup back then but there's more to do.

**Comment by HaakonOfTheVortex** [comment]
I sent Rick and email from his website contact form regarding bug #20.  Arelas and I are back and active and working on things at The Vortex and I will work on verifying bug #20 and see if I can remember bug #21 about the healing potions and money... I vaguely remember it but may just have to find out the hard way.  I discovered it once, I can probably discover it again.  So Arelas told me you had a linux port?  Is it farther along or is .20e still the latest version?  What are you plans for the future of Usurper?  I always wanted to make a sort of Usurper clone and have a bunch of cool ideas but, alas, no time to spend on it.  Maybe we could put our heads together.  What language is this all written in anyway?  Maybe we should all schedule a time for an IRC chat to discuss and do some introductions and such.

**Comment by HaakonOfTheVortex** [comment]
Whoa, blast from the past.  As I recall, the issue was that if you did an
online battle with somebody and defeated the person and took all his
weapons and/or inventory, before the player fight engine sequence was
completely over the loser could join the game from a different node on the
BBS before his data was updated, be online with all his weapons and
inventory, and then the winner could hit the last "enter" key essentially
and both players would have those items then because the loser's data file
was then locked because he was in the game already.  Hope this helps!

**Comment by HaakonOfTheVortex** [comment]
Sadly, I can't remember how the healing potion bug worked....

**Comment by BlaZZZed1980** [comment]
I think with the healing potions, they're talking about the ability to accumulate millions of them by doing the white-water massage, wishing for them at the wishing well, and pickpocketing them from others.  I have personally utilized this one myself just to add an extra hundred or so before doing a hard quest that I wasn't exactly ready for.  There is another thread on this and I believe a viable solution has been proposed.  Unless it's something else entirely.

**Comment by BlaZZZed1980** [comment]
Just curious... are you Lenny from the Lenny's Usurper Page?  Your name of "Haakon" seems very familiar, but I'm not sure where from.  The Lenny's Usurper Page used to have links to lots of statistical data like the drink stat boosts, and the different masters for the various character types, that I'd like to access again.  If it's you, do you know where I can find that stuff now?  And if it's not you, never mind.

**Comment by HaakonOfTheVortex** [comment]
Nope, Lenny was Lenny and still is.  I'm Haakon from the Vortex BBS,
different incarnations but online since roughly 1999 or so.... vortexbbs.com.
I originally posted this list of the 20 fatal Usurper flaws like 15 years
ago... LOL
